docs/ceph-csi-upgrade.md

```diff
@@ -52,12 +52,21 @@ fuse client or rbd-nbd as of now.
 This guide will walk you through the steps to upgrade the software in a cluster
 from v3.0 to v3.1
 
-### snapshot-controller and snapshot Beta CRD
+### Snapshot-controller
 
 Its kubernetes distributor responsibility to install new snapshot
-controller and snapshot beta CRD. more info can be found
+controller and snapshot CRD. more info can be found
 [here](https://github.com/kubernetes-csi/external-snapshotter/tree/master#usage)
 
+#### Snapshot API version support matrix
+
+| Snapshot API version | Kubernetes Version   | Snapshot-Controller + CRDs Version | Sidecar Version        |
+| -------------------- | -------------------- | ---------------------------------- | ---------------------- |
+| v1beta1              | v1.17 =< k8s < v1.20 | v2.x =< snapshot-controller < v4.x | sidecar >= v2.x        |
+| v1                   | k8s >= v1.20         | snapshot-controller >= v4.x        | sidecar >= v2.x        |
```

Comment:
sidecar should also be sidecar >= v4.x? so both will be of the same version

Reply:
Actually, if you see my previous comments above, I was trying to convey that "according to the validation results, If the controller+crds are 4.x and sidecar is 3.x, then we can still use v1 APIs"

from my validation scenario 2 above:
i.e.
```
Scenario 2:
Setup: k8s-1.20 | sidecar v3.x | controller v4.x (with v1 and v1beta1 APIs)

Create few snapshots on {k1.20, sidecar v3.x, controller v3.x} setup
Upgrade controller+crds only from v3.x-> v4.x, try to restore pvc
Create new snapshots with v1beta API (of VolumeSnapshot) and try to restore pvc
Create new snapshots with v1 API (both v1 of VolumeSnapshotClass and VolumeSnapshot) versions and try to restore pvc
```

I can confirm that, if we have the sidecar at a lower version (3.x) and controller +crds ar 4.x, we still can use v1 API.

But @Madhu-1, I will leave that you on what should be the recommended sidecar version at upgrade docs matrix.

Reply:
Agree. but here it is mention as `>=v2.x` maybe change it to 3.x? as we are mentioning to keep both sidecar and controller to the same version I have mentioned v4.x. either change it to 3.x or 4.x anything is fine for me.

Reply:
@Madhu-1

v1beta1 got introduced into external snapshotter with v2.x, see https://github.com/kubernetes-csi/external-snapshotter/blob/master/CHANGELOG/CHANGELOG-2.0.md#api-changes

Let me know what you think, Thanks!

Reply:
v2.x is old now as we have v4 and there we some major performance fixes done in v3.x which are not backported to 2x.